be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -420,23 +424,36 @@ Status SegmentIterator::_try_to_update_ranges_by_runtime_filter() {
             _opts.stats->raw_rows_read);
 }
 
-StatusOr<std::unique_ptr<ColumnIterator>> SegmentIterator::_new_dcg_column_iterator(uint32_t ucid,
-                                                                                    std::string* filename) {
-    // build column iter from delta column group
+StatusOr<std::shared_ptr<Segment>> SegmentIterator::_get_dcg_segment(uint32_t ucid, int32_t* col_index) {
     // iterate dcg from new ver to old ver
     for (const auto& dcg : _dcgs) {
-        int idx = dcg->get_column_idx(ucid);
+        int32_t idx = dcg->get_column_idx(ucid);
         if (idx >= 0) {
             std::string column_file = dcg->column_file(parent_name(_segment->file_name()));
             if (_dcg_segments.count(column_file) == 0) {
                 ASSIGN_OR_RETURN(auto dcg_segment, _segment->new_dcg_segment(*dcg));
                 _dcg_segments[column_file] = dcg_segment;
             }
-            if (filename != nullptr) {
-                *filename = column_file;
+            if (col_index != nullptr) {
+                *col_index = idx;
             }
-            return _dcg_segments[column_file]->new_column_iterator(idx);
+            return _dcg_segments[column_file];
+        }
+    }
+    // the column not exist in delta column group
+    return nullptr;
+}
+
+StatusOr<std::unique_ptr<ColumnIterator>> SegmentIterator::_new_dcg_column_iterator(uint32_t ucid,
+                                                                                    std::string* filename) {
+    // build column iter from delta column group
+    int32_t col_index = 0;
+    ASSIGN_OR_RETURN(auto dcg_segment, _get_dcg_segment(ucid, &col_index));
```

Comment:
Does the bitmap index exist in the main segment, or in the delta column group?

Reply:
The old columns's bitmap index exist in the main segment, and the new columns(after update)'s bitmap index exist in the delta column group.